railties/lib/rails/commands/server/server_command.rb

```diff
@@ -238,12 +256,50 @@ def pid
         end
 
         def self.banner(*)
-          "rails server [puma, thin etc] [options]"
+          "rails server [thin/puma/webrick] [options]"
         end
 
         def prepare_restart
           FileUtils.rm_f(options[:pid]) if options[:restart]
         end
+
+        def deprecated_positional_rack_server(value)
+          if value
+            ActiveSupport::Deprecation.warn(<<-MSG.squish)
+              Passing the Rack server name as a regular argument is deprecated
+              and will be removed in the next Rails version. Please, use the -u
+              option instead.
+            MSG
+            value
+          end
+        end
+
+        def rack_server_suggestion(server)
+          if server.in?(RACK_SERVERS)
+            <<~MSG
+              Could not load server "#{server}". Maybe you need the add it to the Gemfile?
+
+                gem "#{server}"
+
+              Run `rails server --help` for more options.
+            MSG
+          else
+            suggestions = Rails::Command::Spellchecker.suggest(server, from: RACK_SERVERS).map(&:inspect)
+
+            <<~MSG
+              Could not find server "#{server}". Maybe you meant #{suggestions.first} or #{suggestions.second}?
+              Run `rails server --help` for more options.
+            MSG
```

Comment:
For another time: I could totally see us enriching Thor's argument/option with a `:suggestions` to get messages like this for free. E.g.:

```ruby
class_option :using, aliases: "-u", type: :string, suggestions: RACK_SERVERS,
  desc: "Specifies the Rack server used to run the application (thin/puma/webrick).", banner: :name
```